packages/native/src/useLinking.tsx

```diff
@@ -296,12 +296,25 @@ export function useLinking(
             focusedRoute.name === route.name &&
             isEqual(focusedRoute.params, route.params)
           ) {
-            return route.path;
+            return route.path + location.hash;
           }
         }
       }
 
-      return getPathFromStateRef.current(state, configRef.current);
+      const [previousFocusedState] = findMatchingState(
+        previousStateRef.current,
+        state
+      );
+
+      const lastScreen =
+        previousFocusedState?.routes?.[
+          previousFocusedState?.routes?.length - 1
+        ];
+      // @ts-expect-error key is ommited for focused route
```

Comment:
can not get it why do we have omitted 'key' prop in return type findFocusedRoute - but it present there